File: frontend/src/classe/Referent.js
export class Referent {
  constructor(idReferent, nom, prenom, email) {
    this._idReferent = idReferent;
    this._nom = nom;
    this._prenom = prenom;
    this._email = email;
  }

  get idReferent() { return this._idReferent; }
  set idReferent(value) { this._idReferent = value; }

  get nom() { return this._nom; }
  set nom(value) { this._nom = value; }

  get prenom() { return this._prenom; }
  set prenom(value) { this._prenom = value; }

  get email() { return this._email; }
  set email(value) { this._email = value; }
}

File: backend/src/main/java/isis/projet/backend/entity/Referent.java
package isis.projet.backend.entity;

import com.fasterxml.jackson.annotation.JsonManagedReference;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;
/**
 * Entité représentant un référent.
 * Un référent peut valider plusieurs actions.
 */
@Entity
@Table(name = "referent")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Referent {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id_referent")
    private Integer idReferent;

    @Column(name = "nom")
    private String nom;

    @Column(name = "prenom")
    private String prenom;

    @Column(name = "email")
    private String email;

    @OneToMany(mappedBy = "referent", cascade = CascadeType.ALL)
    @JsonManagedReference(value = "referent-participations")
    private List<Participe> participationsValidees;
}